File: bindings/dotnet/src/Zlink/Runtime/Sockets/XSubSocket.cs
// SPDX-License-Identifier: MPL-2.0

namespace Systems.Zlink;

internal sealed class XSubSocket : SubscriberSocketBase, IXSubSocket
{
    public XSubSocket(Context context)
        : base(context, SocketType.XSub)
    {
        Options = new SubSocketOptions(this);
    }

    public new SubSocketOptions Options { get; }
}

File: bindings/node/src/zlink/runtime/sockets/xsub_socket.ts
// SPDX-License-Identifier: MPL-2.0

import { SubSocketOptions } from './socket_options';
import {
  SubscriberSocket,
} from './socket_operations';
import type { RuntimeContext as Context } from '../core/context';
import { SocketType as NativeSocketType } from '../../contracts/sockets/socket_constants';

export class XSubSocket extends SubscriberSocket {
  readonly options: SubSocketOptions;
  constructor(ctx: Context) { super(ctx, NativeSocketType.XSUB); this.options = SubSocketOptions.create(this); }
}
